pkg/controller/component/storaged_failover.go

```diff
@@ -244,3 +289,151 @@ func (s *storagedFailover) checkPendingPod(nc *v1alpha1.NebulaCluster) error {
 	}
 	return nil
 }
+
+// balanceStorageLeader balances leaders across storage nodes
+func (s *storagedFailover) balanceStorageLeader(nc *v1alpha1.NebulaCluster) error {
+	if len(nc.Status.Storaged.FailureHosts) == 0 {
+		klog.Infof("no failure hosts in storaged cluster [%s/%s], skip leader balance", nc.GetNamespace(), nc.GetName())
+		return nil
+	}
+
+	options, err := nebula.ClientOptions(nc, nebula.SetIsMeta(true))
+	if err != nil {
+		return err
+	}
+
+	endpoints := []string{nc.GetMetadThriftConnAddress()}
+	metaClient, err := nebula.NewMetaClient(endpoints, options...)
+	if err != nil {
+		klog.Errorf("create meta client failed: %v", err)
+		return err
+	}
+
+	defer func() {
+		if err := metaClient.Disconnect(); err != nil {
+			klog.Errorf("disconnect meta client failed: %v", err)
+		}
+	}()
+
+	spaces, err := metaClient.ListSpaces()
+	if err != nil {
+		return err
+	}
+
+	if len(spaces) > 0 && nc.Status.Storaged.BalancedSpaces == nil {
+		nc.Status.Storaged.BalancedSpaces = make([]int32, 0, len(spaces))
+	}
+
+	for _, space := range spaces {
+		if contains(nc.Status.Storaged.BalancedSpaces, *space.Id.SpaceID) {
+			continue
+		}
+		if err := metaClient.BalanceLeader(*space.Id.SpaceID); err != nil {
+			return err
+		}
+		nc.Status.Storaged.BalancedSpaces = append(nc.Status.Storaged.BalancedSpaces, *space.Id.SpaceID)
+	}
+
+	// Reset balance status
+	nc.Status.Storaged.BalancedSpaces = nil
+	nc.Status.Storaged.LastBalanceJob = nil
+
+	return nil
+}
+
+// check if there are more than 2 failure hosts in the same part
+func (s *storagedFailover) checkPeerExist(nc *v1alpha1.NebulaCluster, failureHosts []string) (bool, error) {
```

Comment:
Want to change the name of this func? What you mean is check if there are more than 2 failed hosts in the same part. But what this function name mean is checking if the peer exists.